dev/io.openliberty.check_dhe_fat/fat/src/io/openliberty/check_dhe/VerifyDHETest.java

```diff
@@ -0,0 +1,177 @@
+
+        for (GAVEntry entry : gavSet) {
+            // Verify that the artifact directory exists
+            String baseURL = entry.getBaseURL(dheServerRootURL);
+            System.out.println("################################################################################");
+            System.out.println("Testing " + baseURL);
+
+            URL url = new URL(baseURL);
+            try {
+                HttpURLConnection httpConn = HttpUtils.getHttpConnection(url, 200, 10);
+                String data = HttpUtils.readConnection(httpConn);
+                System.out.println(data);
+
+                // Check that references to the jar (or war) and pom file exist
+                String jarFileName = entry.getArtifactName() + "-" + entry.getVersionName() + "." + entry.getFileType();
+                String expectedJarAnchor = "<a href=\"" + jarFileName + "\">" + jarFileName + "</a>";
+
+                String pomFileName = entry.getArtifactName() + "-" + entry.getVersionName() + ".pom";
+                String expectedPomAnchor = "<a href=\"" + pomFileName + "\">" + pomFileName + "</a>";
+
+                Assert.assertTrue("DHE does not have a jar file for " + entry, data.contains(expectedJarAnchor));
+
+                if (!excludeFromPomCheckSet.contains(entry)) {
+                    Assert.assertTrue("DHE does not have a pom file for " + entry, data.contains(expectedPomAnchor));
+                }
+            } catch (java.lang.AssertionError ae) {
+                assertionErrorList.add(ae);
+                failedGAVEntries.add(entry);
+            } catch (Throwable t) {
+                assertionErrorList.add(new java.lang.AssertionError("Failed to read resource at " + baseURL, t));
+                failedGAVEntries.add(entry);
+
+                StringBuilder sb = new StringBuilder();
+                sb.append("Failed to locate GAV asset ").append(entry).append(" at ").append(baseURL);
+                sb.append("\n").append("  Error: ").append(t.toString());
+                assertionErrorList.add(new java.lang.AssertionError(sb.toString()));
+            }
+        }
+
+        if (failedGAVEntries.size() > 0) {
+            StringBuilder sb = new StringBuilder();
+            sb.append(failedGAVEntries.size()).append(" Failed GAV Entries:\n");
+            for (GAVEntry e : failedGAVEntries) {
+                sb.append("  ").append(e).append("\n");
+            }
+            assertionErrorList.add(new java.lang.AssertionError(sb.toString()));
+        }
+
+        if (assertionErrorList.size() > 0) {
+            StringBuilder sb = new StringBuilder();
+            sb.append(assertionErrorList.size() + " Failed Assertions have been detected:\n");
+            for (java.lang.AssertionError ae : assertionErrorList) {
+                sb.append("######################################################################\n");
+                sb.append(exceptionToString(ae));
```

Comment:
In the event that mutliple artifacts are not found this exception message could get large very quickly -- probably 100 lines per failure. Instead, could we just include the first line (i.e. exception message) of each failure? The stack traces themselves are mostly irrelevant because they'll all be the same.

So perhaps an exception message that looks like:
```
3 Failed GAV Entries:
  com:foo:1.0
  com:foo:2.0
  com:foo:3.0
at VerifyDHETest.java (line X)
at <etc..>
```